wordpress-mobile/WordPress-iOS
# Second notification prompt for users that have not set notifications permissions

Ref #14202

This PR adds a second Notifications Permissions prompt for those users who chose "Not Now" on the inline prompt in the 'Notifications' tab. The alert will look like in the picture below:

<img src=https://user-images.githubusercontent.com/34376330/88204555-0a726180-cc11-11ea-9295-a15b6e1a5fb2.png width =300>

To test:
pre requisites:

- a WordPress.com account on a device
- another WordPress.com account (on another device or simulator) that can trigger notifications on the first account (e.g. follow a site from the first account, where you can like posts or comment on them), or any other way you could think of, to send notifications to the first account

Test 1
1. Fresh install the app on the device and login with the first account
2. Go to 'Notifications' and, when prompted for permissions, tap "Not Now"
3. Visit 'Notifications' 5 more times until the inline prompt shows
4. On the inline prompt, tap "Not Now"
5a. While on 'Notifications', trigger 10 notifications (e.g. like or comment on a post with the second account)and verify that a second alert with title "Find out faster with push notifications" shows up (see example above)
5b. While on a tab other than 'Notifications', trigger 10 or more notifications, then go to 'Notifications' and verify that a second alert with title "Find out faster with push notifications" shows up (see example above)

Test2:
- go to steps 1 to 4 in Test 1 (or any subset of it) with any version of the app
- Update the app to this branch and proceed with the remaining steps 

##### Test tips:
1. To avoid waiting for 10 notifications, you can temporarily set the treshold to a lower value. This can be done tweaking 'Constants.secondNotificationsAlertTreshold` in `NotificationsAlertViewController.swift`
2. To re-test without re-installing, you can reset Userdefaults values as follows:

- pause the debugger
- to reset the second alert notifications counter, in the lldb console, type:
`e -l swift -- import WordPress; UserDefaults.standard.secondNotificationsAlertCount = 0`
- to reset the inline prompt value (so that it shows up again) in the lldb console, type:
`e -l swift -- import WordPress; UserDefaults.standard.notificationPrimerInlineWasAcknowledged = false`
- Relaunch the app and the values should be reset

3. The test will work as long as the push notifications permissions are not yet set. If at any moment before steps 5 you set the permissions (by tapping on "Allow push notifications" and then allowing or denying them in the system), the second alert will not show (regardless resetting the user defaults). In this case, you'll need to do a fresh install again to re-test.

PR submission checklist:
